src/twisted/trial/_dist/test/test_worker.py

```diff
@@ -331,19 +331,49 @@ def test_childDataReceived(self):
         L{AMP} protocol if the right file descriptor, otherwise forwards to
         C{ProcessProtocol.childDataReceived}.
         """
-        localWorker = self.tidyLocalWorker(FakeAMProtocol(), ".", "test.log")
+        localWorker = self.tidyLocalWorker(SpyDataLocalWorkerAMP(), ".", "test.log")
         localWorker._outLog = BytesIO()
         localWorker.childDataReceived(4, b"foo")
         localWorker.childDataReceived(1, b"bar")
         self.assertEqual(b"foo", localWorker._ampProtocol.dataString)
         self.assertEqual(b"bar", localWorker._outLog.getvalue())
 
+    def test_unicodeLogFileUTF8(self):
+        """
+        L{LocalWorker} write the log data in UTF-8 encoding regardless of the
+        default python encoding.
+        """
+        amp = SpyDataLocalWorkerAMP()
+        tempDir = FilePath(self.mktemp())
+        logFile = tempDir.child("test.log")
+
+        def getLatin1(do_setlocale: bool = True) -> str:
+            """
+            A replacement for L{locale.getpreferredencoding} that always
+            returns the Latin-1 encoding.
+            """
+            return "latin-1"
+
+        self.patch(_bootlocale, "getpreferredencoding", getLatin1)
```

Comment:
I am happy to write this test in another way.
I have observed that patching just `locale` was not triggering the expected TDD initial test failure.